contracts/core/SequencerSoloClient-ancestorProof.sol

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ * Copyright 2024, Polymer Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity 0.8.15;
+
+import {Ics23Proof} from "../interfaces/IProofVerifier.sol";
+import {ISignatureVerifier} from "../interfaces/ISignatureVerifier.sol";
+import {ILightClient, LightClientType} from "../interfaces/ILightClient.sol";
+import {L1Block} from "optimism/L2/L1Block.sol";
+import {console2} from "forge-std/console2.sol";
+
+/**
+ * @title SubfinalityLightClient
+ * @author Polymer Labs
+ * @dev This specific light client implementation uses the same client that is used in the op-stack
+ */
+contract SequencerSoloClientAncestorProof is ILightClient {
```

Comment:
is the plan to add ancestor proof check logic later? it seems the same as `SequencerSoloClientCheckpoint`

Reply:
not necessarily, these were just contracts I was playing around with that I didn't mean to include in this PR - sorry for the confusion!